python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -198,20 +217,51 @@ def getWNScriptDownloadCommand(self, indent):
 
         return script
 
+    def _mkdir(self, massStoragePath, force_=False):
+        """
+        Creates a folder on the mass Storage corresponding to the given path
+        Args:
+            massStoragePath (str): This is the path we want to make if it doesn't exist.
+        """
+
+        massStorageConfig = getConfig('Output')['MassStorageFile']['uploadOptions']
+        mkdir_cmd = massStorageConfig['mkdir_cmd']
+        ls_cmd = massStorageConfig['ls_cmd']
+
+        # create the last directory (if not exist) from the config path
+        pathToDirName = os.path.dirname(massStoragePath)
+        dirName = os.path.basename(massStoragePath)
+
+        directoryExists = False
+
+        (exitcode, mystdout, mystderr) = self.execSyscmdSubprocess('%s %s' % (ls_cmd, escapeWhiteSpace(pathToDirName)))
+        if exitcode != 0:
+            #self.handleUploadFailure(mystderr, '1) %s %s' % (ls_cmd, pathToDirName))
+            #raise GangaException(mystderr)
+            directoryExists = True
```

Comment:
If the dir **doesn't** exist, exitcode will be non-zero. Surely at that point it should try to create it? As it stands this will never actually create the directory.